crates/aptos-crypto/src/test_utils.rs

```diff
@@ -279,7 +279,7 @@ impl std::io::Write for TestAptosCryptoHasher {
         Ok(())
     }
 }
-#[cfg(any(test, feature = "fuzzing"))]
+// #[cfg(any(test, feature = "fuzzing"))]
```

Comment:
too long; **don't** read: Because I am trying to document our BLS library so people don't misuse it. As part of documenting (see [this](https://github.com/aptos-labs/aptos-core/pull/2659/files#diff-c65467f20822c93ea9f34d6703f57b5e0c8d312289840725e9785b0aae9d4706R133)), I have to show how to sign a Rust struct by deriving the `CryptoHasher` and `BCSCryptoHash` traits. 

But "naturally", our current macros for this are in `aptos-crypto-derive` and are completely unusable if called from the `aptos-crypto` crate because, at some point, they generate a `use aptos_crypto::<bla>`. This works fine when called from other crates, but not from `aptos-crypto`, where the macros should generate a `use crate::<bla>` instead.

As a result, I could not give any example in this crate... **unless** I use the `TestAptosCrypto` struct which, also "naturally" manually implements `CryptoHasher` and `BCSCryptoHash` instead of using the `aptos-crypto-derive` macros.

...hence, I had to make `TestAptosCrypto` publicly-visible.

If this is a big problem, please suggest an alternative? In the past, I figured out how to actually make `use aptos_crypto::<bla>` work from the `aptos-crypto` crate by doing some weird aliasing in Rust. However, that messes up auto-completion in CLion: everything appears twice.

Reply:
Specifically, as a hotfix I did:

```
// We need to add this here if we want to use aptos-crypto-derive's CryptoHasher and BCSCryptoHasher
// macros because these macros generate a `use aptos_crypto::hash::CryptoHash` line which will fail
// inside this crate (i.e., it would need to be a `use crate::hash::CryptoHash` line instead).
extern crate self as aptos_crypto;
```